File: old/Epic Game/1.rb
x,y,z=gets.chomp.split()
x = x.to_i
y = y.to_i
z = z.to_i
point = 0
while true
	if point==0
		z-=x.gcd(z)
		if z<=0
			puts 0
			break
		end
		point+=1
	end
	if point==1
		z-=y.gcd(z)
		if z<=0
			puts 1
			break
		end
		point-=1
	end

end

File: old/Epic Game/1.py
from fractions import gcd
x,y,z = map(int, raw_input().split())
point = 0
while z>0:
	if point==0:
		z-=gcd(x,z)
		if z<=0:
			print 0
		else:
			point+=1
		
	elif point==1:
		z-=gcd(y,z)
		if z<=0:
			print 1
		else:
			point-=1